app/controllers/Controller.scala

```diff
@@ -52,12 +55,65 @@ class Controller @Inject() (
     implicit request: Request[_] =>
       af.authorised() {
 
-        Future.successful(Ok("manageOrTrack"))
+        manageOrTrackView(vrn, manageOrTrackForm.fill(ManageOrTrack(None, vrn.value)))
+
+      }.recoverWith {
+        case e: AuthorisationException => authorisationException(e)
+      }
+  }
+
+  def manageOrTrackSubmit(): Action[AnyContent] = Action.async {
+    implicit request =>
+      af.authorised() {
+        val mtFormOptional = manageOrTrackForm.bindFromRequest()
+        mtFormOptional.value match {
```

Comment:
Please refactor it using `fold[R](hasErrors: Form[T] => R, success: T => R)` method on form.